### PR TITLE
WGLC changes in Privacy Considerations section.

### PR DESCRIPTION
**Pull request by rohan-wire**
WGLC comments on Security Considerations Section

- Added discussion about significance of `basic` vs. `x509` credential types.
- Moved a paragraph about transport security in general into that section.
- Made text with list of metadata consistent with the metadata that is actually present in the protocol now.
- Moved Metadata Protection for Unencrypted Group Operations heading to just before the two paragraphs that actually discuss it.
- Added subsection "Integrity and Authentication of Custom Metadata" before the discussion about adding custom metadata to the AAD field.
- Noted that a websocket connection directly to the DS can be used instead of mobile host platform (ex: Apple, Google) native notifications in some environments.
- Fixed some typos.

### CODE REVIEW

**Comment by rohan-wire** [comment]
The automatic checks are failing claiming make lint failed due to trailing whitespace.
make lint is running fine in my local copy. I also did a manual search for trailing whitespace and could not fine any. Could someone please kick github actions?

File: draft-ietf-mls-architecture.md [hunk 1]
```diff
@@ -1337,6 +1358,17 @@ clients, it can provide stale keys. This does not inherently lead to compromise
 of the message stream, but does allow it to attack forward security to a limited
 extent. This threat can be mitigated by having initial keys expire.
 
+Initial keying material (KeyPackages) using the `basic` Credential type
+is more vulnerable to replacement by a malicious or compromised DS, as
+there is no built-in cryptographic binding between the identity and the
+public key of the client.
```
Comment (Bren2010) [inline draft-ietf-mls-architecture.md:1364]:
```suggestion
public key of the client.
```

File: draft-ietf-mls-architecture.md [hunk 2]
```diff
@@ -1337,6 +1358,17 @@ clients, it can provide stale keys. This does not inherently lead to compromise
...
+Initial keying material (KeyPackages) using the `basic` Credential type
+is more vulnerable to replacement by a malicious or compromised DS, as
+there is no built-in cryptographic binding between the identity and the
+public key of the client. 
+
+> **RECOMMENDATION:**
+> Prefer a Credential type in KeyPackages which includes a strong
+> cryptographic binding between the identity and its key (for example
+> the `x509` Credential type). When using the `basic` Credential type
+> take extra care to verify the identity (typically out-of-band).
```
[the remark this answers is not in the document]
Comment (bifurcation) [reply draft-ietf-mls-architecture.md:1370]: If you look at most E2E-encrypted messaging applications today (e.g., Signal, WhatsApp, Wire), there's no binding of identity material to key material that is safe from DS manipulation, aside from manual verification.  If these systems were to move to MLS without inventing a new identity system, `basic` would be the obvious way to do it.

File: draft-ietf-mls-architecture.md [hunk 3]
```diff
@@ -1419,21 +1452,33 @@ One thing for which the MLS Protocol is designed for is to make sure that all
 clients know who is in the group at all times. This means that - if all Members
 of the group and the Authentication Service are honest - no other parties than
 the members of the current group can read and write messages protected by the
-protocol for that Group.
+protocol for that Group. Note that when BasicCredential is used, the
+Authentication Service also needs an out-of-band mechanism to verify the
+identity asserted in the Credential.
```
Comment (bifurcation) [inline draft-ietf-mls-architecture.md:1457]:
```suggestion
protocol for that Group. Note that when a `basic` Credential is used, the
Authentication Service also needs an out-of-band mechanism to verify the
identity asserted in the Credential.
```

File: draft-ietf-mls-architecture.md [hunk 4]
```diff
@@ -1419,21 +1452,33 @@ One thing for which the MLS Protocol is designed for is to make sure that all
...
+designs, a private or centralized authority can be trusted to generate or
+validate signature keypairs used in the MLS protocol. This is typically the
+case in some of the biggest messaging infrastructures.
+
+Details about how to verify the identity of a client depend on the MLS
+Credential type used. For example, cryptographic verification of credentials
+can be largely performed autonomously on the clients for the `x509`
+Credential type. In contrast, when MLS clients use the `basic` Credential type,
+a larger degree of trust must be placed in a (likely) centralized
+authentication resource.
```
Comment (bifurcation) [inline draft-ietf-mls-architecture.md:1470]:
```suggestion
authentication resource, or on out-of-band processes such as manual verification.
```

**Review by bifurcation — APPROVED** [review]

**Pull request merged**